lib/ash/filter/filter.ex

```diff
@@ -2708,46 +2708,43 @@ defmodule Ash.Filter do
        when is_atom(field) or is_binary(field) do
     cond do
       rel = relationship(context, field) ->
-        with relation_type when relation_type != :many_to_many <- rel.type,
-             true <- !Map.get(rel, :no_attributes),
-             [pk] when pk == rel.destination_attribute <-
-               Ash.Resource.Info.primary_key(rel.destination),
-             attr <- attribute(%{public?: true, resource: rel.source}, rel.source_attribute),
+        with rel_type when rel_type != :many_to_many <- rel.type,
+             dest <- rel.destination,
```

Comment:
We still need the `true <- !Map.get(real, :no_attributes)`.

Reply:
Right, I must have lost it somewhere on the way.